src/citrine/jobs/job.py

```diff
@@ -96,7 +97,12 @@ def _poll_for_job_completion(session: Session,
         job_id = job.job_id
     else:
         job_id = job  # pragma: no cover
-    path = format_escaped_url('projects/{}/execution/job-status', project_id)
+    if project_id is None and team_id is None:
+        raise TypeError('Either a project_id or team_id must be provided to poll a job.')
+    if team_id is not None:
+        path = format_escaped_url('teams/{}/execution/job-status', team_id)
+    else:
+        path = format_escaped_url('projects/{}/execution/job-status', project_id)
```

Comment:
I need to check with Pablo. Based on his comment above, it sounds like the intent is for the project-level jobs endpoint to be deprecated.

Reply:
If the team_id is known we should call the team API. I'm unsure if we have guarantees of the team_id to be available here. But yes, ideally we should be moving to the new pattern. As a side note, the project-based job ID is not going to be deprecated because of the data manager, it will be for another reason (having a homogenous API for dealing with jobs).